### PR TITLE
Feat/mds 3859 add multiple phone numbers and emails to parties

### PR DESCRIPTION
**Pull request by dragancolic-bcgov**
# Main

- The focus was on adding additional two phones (secondary and tertiary) and email (secondary) to the contact forms.

# Other

- What other side fixes, tweaks did you add on the side?
- N/A

# How to test

- in Core navigate to: Contacts -> Add New Contact.
- in Core navigate to Contacts, click on any contact then Update Contact.
- in Core list mines, click on any mine, under Mine Information click on Contacts -> Add new Contact.

# Notes

- Any notes/screenshots for this PR?
- N/A

### CODE REVIEW

**Comment by exzizt** [comment]
Don't forget this AC bullet: _View new emails and phone numbers on the UI in all spots it is currently being shown (Party cards, party profile, contact table, etc.)_

File: migrations/sql/V2021.09.02.03.00__add_phone_and_email_columns_to_party.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+ALTER TABLE party ADD COLUMN phone_no_sec character varying(12) NULL;
```
Comment (exzizt) [inline migrations/sql/V2021.09.02.03.00__add_phone_and_email_columns_to_party.sql]: Can you add "IF NOT EXISTS" to each of these? Migrations are generally not re-runnable but sometimes this causes issues locally. We also don't want to enforce character lengths, or any other types of similar constraints, at the DB level; something that was decided upon very early in the project. We only do this for some things like code tables now.

File: services/core-api/app/api/parties/party/models/party.py [hunk 2]
```diff
@@ -150,15 +174,25 @@ def create(cls,
                email=None,
                first_name=None,
                phone_ext=None,
-               add_to_session=True):
+               add_to_session=True,
```
Comment (exzizt) [inline services/core-api/app/api/parties/party/models/party.py]: Can you please move the add_to_session paramater back to be the last parameter?

File: services/core-api/app/api/parties/party/resources/party_list_resource.py [hunk 3]
```diff
@@ -129,7 +139,13 @@ def post(self):
             data.get('party_type_code'),
             email=data.get('email'),
             first_name=data.get('first_name'),
-            phone_ext=data.get('phone_ext'))
+            phone_ext=data.get('phone_ext'),
+            add_to_session=True,
```
Comment (exzizt) [inline services/core-api/app/api/parties/party/resources/party_list_resource.py]: And this one

**Comment by exzizt** [comment]
Can you also please fill out the PR doc template and provide the necessary screenshots of UI changes?

**Comment by dragancolic-bcgov** [comment]
> Can you also please fill out the PR doc template and provide the necessary screenshots of UI changes?

Sorry Luke, my PR is still not ready for review. I've just noticed that we automatically add a flag 'ready for review' to our PRs and that wasn't my initial intention...
Anyway, I'm working on refactoring and will let you know when it becomes ready for review.

Thanks.

**Comment by exzizt** [comment]
> > Can you also please fill out the PR doc template and provide the necessary screenshots of UI changes?
> 
> Sorry Luke, my PR is still not ready for review. I've just noticed that we automatically add a flag 'ready for review' to our PRs and that wasn't my initial intention...
> Anyway, I'm working on refactoring and will let you know when it becomes ready for review.
> 
> Thanks.

Ok my bad, I thought you said during standup it was!

**Comment by dragancolic-bcgov** [comment]
> > > Can you also please fill out the PR doc template and provide the necessary screenshots of UI changes?
> > 
> > 
> > Sorry Luke, my PR is still not ready for review. I've just noticed that we automatically add a flag 'ready for review' to our PRs and that wasn't my initial intention...
> > Anyway, I'm working on refactoring and will let you know when it becomes ready for review.
> > Thanks.
> 
> Ok my bad, I thought you said during standup it was!

No worries. ;) Thanks anyway...

File: migrations/sql/V2021.09.02.03.00__add_phone_and_email_columns_to_party.sql (hunk 1 above)
Comment (areyeslo) [reply migrations/sql/V2021.09.02.03.00__add_phone_and_email_columns_to_party.sql]: You might want to take a look at \mds\migrations\sql\V2021.02.24.12.59__update_party_and_address_tables_remove_length_limitations.sql
for avoiding character lengths in the phone numbers.

**Review by areyeslo — APPROVED** [review]
Well done Dragan :). Thanks for making those changes. Ready to be merged.

**Comment by dragancolic-bcgov** [comment]
> Well done Dragan :). Thanks for making those changes. Ready to be merged.

Thank you for reviewing it, Arturo! :)

**Pull request merged**